### DISCUSSION TITLE
Why do DuckDNS and NGINX add-ons have conflicting requirements for /config.yaml ?

### DISCUSSION THREAD

**Discussion by kmil4** [llamafilm/tesla-http-proxy-addon#63] [Q&A]
I'm trying to transition from the Tesla Owner API to the Fleet API. My environment is a RP4+ running HAOS. I'm subscribed to Nabu Casa for external access but configured DuckDNS to try and use this integration.

I've read the docs carefully and configured add-ons for DuckDNS, LetsEncrypt, NGINX, and Tesla. Using SambaShare in Windows I can see `/nginx_proxy` and `/tesla` under my `/share` directory. I can even see the public key in Samba, but FWIW I get a 404 when trying to reach `https://tesla.[mydomain].duckdns.org/.well-known/appspecific/com.tesla.3p.public-key.pem`

But every time I try to start the Tesla Add-On, I get an error after a few seconds: 
```
...
[16:17:01] INFO: Found existing keypair
[16:17:01] INFO: Testing public key...
[16:17:06] FATAL: Fix public key before proceeding.
```

I've been scouring my logs and the documentation to try and find a solution. A lot of tinkering with add-on configs but I always end up stuck in this spot. 

One thing has me scratching my head. The first screenshot is from DuckDNS docs. The 2nd screenshot is from NGINX docs. These directly conflict... what am I missing? (I also know this might not make a difference, I'm not very knowledgeable in this space)

DuckDNS docs...
![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/91843865/bcc25b3b-cd48-473b-b5b5-ca6c48d8ad66)

NGINX docs...

![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/91843865/314de263-183b-4447-8da2-b84bbeaac605)

**Comment by llamafilm** [comment]
It's a choice, whether you want Home Assistant core to provide the HTTPS server, or use Nginx to run an HTTPS reverse proxy.  Any time you expose a server publicly to the internet, it's generally a good idea to use a server that is widely used and battle tested.  That's why most software projects, including Home Assistant, recommend running behind a reverse proxy.  See here: https://www.home-assistant.io/integrations/http/
> **ssl_certificate** string (optional)
> Path to your TLS/SSL certificate to serve Home Assistant over a secure connection. If using the [Let’s Encrypt add-on](https://github.com/home-assistant/addons/tree/master/letsencrypt) this will be at /ssl/fullchain.pem. We recommend to use the [NGINX add-on](https://github.com/home-assistant/addons/tree/master/nginx_proxy) instead of using this option.

**Reply by kmil4** [reply]
Really appreciate your response. 

This clarifies a few things. Seems like my use of the LetsEncypt addon is redundant. I'll try and simplify things

**Reply by llamafilm** [reply]
That’s not what I meant to suggest. Lets Encrypt is nice to have because it creates and renews certificates for you semi-annually.

The redundant piece here is running both nginx and Nabu Casa. It’s up to you if you want to cancel Nabu Casa and just use nginx. But it should be fine to run both.

Configure nginx as described in your screenshot. Then check the nginx config page and make sure it shows `active: true` which allows it to read additional config files like the one created by this addon.

Test your domain using some dns checker website to make sure it resolves to your public IP.

If you previously ran this addon and made a mistake, you might have a bad config file under /share/nginx_proxy. Check that file to make sure it shows the correct domain.

**Reply by IanK6449** [reply]
I am still receiving the same error even after defining a new duckdns domain name, installing a new cert and then reinstalling NGINX and the Tesla HTTP Proxy. No idea what the problem is but I have pretty much tried everything now. If I define the same domain name in the Tesla Proxy add-on as in the NGINX add-on I receive a warning about a duplicate domain being detected but this being ignored. So, I suspect the problem lies there somewhere but I have no idea how to fix it. Thank you anyway for putting this add-on together. Hopefully it will work for me one day.

**Reply by IanK6449** [reply]
I also receive the following error when trying to access the public cert after the NGINX configuration step:

2024/03/27 19:11:36 [error] 150#150: *50 connect() failed (111: Connection refused) while connecting to upstream, client: 192.168.2.1, server: weeratowers.duckdns.org, request: "GET /api/websocket HTTP/1.1", upstream: "http://172.30.32.1:8123/api/websocket", host: "weeratowers.duckdns.org"

So, I am wondering whether it is a problem with the NGINX setup and the internal IP addresses so perhaps the trusted proxies are incorrect for my setup in configuration.yaml. I used the trusted proxy listed in the instructions but am not sure how to check whether this is the correct one.
